firmware/library/L2_Utilities/utility_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include "L2_Utilities/utility.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+TEST_CASE("Testing Random Functions", "[random]")
+{
+    SECTION("SetRandomSeed Consistency")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = std::rand() % 100;
+        CHECK(randomtester1 == randomtester2);
+    }
+
+    SECTION("Random")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = utility::Random(0, 99);
+        CHECK(randomtester1 == randomtester2);
+    }
+}
+
+TEST_CASE("Testing Bit Manipulations", "[Bit Manipulation methods]")
+{
+    SECTION("BitExtract")
+    {
+        CHECK(utility::BitExtract(15, 2, 3) == 7);
+        CHECK(utility::BitExtract(15, 3, 3) == 7);
+        CHECK(utility::BitExtract(15, 31, 32) == 15);
+        CHECK(utility::BitExtract(15, 0, 1) == 1);
+        CHECK(utility::BitExtract(15, 31, 28) == 0);
+    }
+
+    SECTION("BitPlace")
+    {
+        CHECK(utility::BitPlace(15, 2, 7, 3) == 15);
+        CHECK(utility::BitPlace(15, 4, true, 1) == 31);
+        CHECK(utility::BitPlace(15, 3, false, 1) == 7);
+        CHECK(utility::BitPlace(0xF0F0F0F0, 31, 0x0F0F0F0F, 32) == 0xFFFFFFFF);
```

Comment:
The behavior of bit place should be to replace the set of bits with a new set of bits for a specific width and position. For example, lets say I have the hex value 0x12345678, and I want to place DEAD in the middle (8 bits the left), the result should be 0x12DEAD78. It shouldn't simply or the bits together. Otherwise, a simple bit set function would be a better implementation for this.